lib/compeon/rack_tools/token.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'jwt'
+require 'open-uri'
+
+module Compeon
+  module RackTools
+    module Token
+      class ParseError < RuntimeError; end
+      class InvalidTokenKindError < ParseError; end
+
+      AccessToken = Struct.new(:client_id, :role, :user_id, keyword_init: true)
+      AuthorizationToken = Struct.new(:client_id, :user_id, :redirect_uri, keyword_init: true)
+
+      class << self
+        def parse_access_token(token)
+          access_token = parse_token(token, 'access')
+
+          AccessToken.new(
+            client_id: access_token.fetch('cid'),
+            role: access_token.fetch('role'),
+            user_id: access_token.fetch('uid')
```

Comment:
```suggestion
            user_id: access_token.fetch('uid'),
            token: token
```
It's help full to use the original token string for example to make act as user request to other API endpoints